### PR TITLE
(AB#145508) Update config conceptual doc

### PR DESCRIPTION
**Pull request by michaeltlombardi**
# PR Summary

This change drafts conceptual documentation for DSC Configuration documents in DSCv3, replacing the temporary note about the documentation.

## PR Checklist

- [x] **Descriptive Title:** This PR's title is a synopsis of the changes it proposes.
- [x] **Summary:** This PR's summary describes the scope and intent of the change.
- [x] **Contributor's Guide:** I have read the [contributors guide][contrib].
- [x] **Style:** This PR adheres to the [style guide][style].

[contrib]: https://learn.microsoft.com/powershell/scripting/community/contributing/overview
[style]: https://learn.microsoft.com/powershell/scripting/community/contributing/powershell-style-guide

### CODE REVIEW

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md [hunk 1]
```diff
@@ -6,7 +6,201 @@ description: >-
...
 ---
+
 # DSC Configuration Documents
 
+<!-- markdownlint-disable MD013 -->
+
+In Microsoft's Desired State Configuration (DSC) platform, DSC configuration documents declare the
+desired state of a system as data files. Configuration documents define a collection of
+[DSC Resource][01] instances to describe what the desired state should be, not how to put the
+system into that state. The DSC Resources handle the "how" for the configuration document's
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:17]:
```suggestion
system into that state. The DSC Resources handle the _how_ for the configuration document
```

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md [hunk 2]
```diff
@@ -6,7 +6,201 @@ description: >-
...
+
 > [!NOTE]
-> The team is rewriting this document for DSCv3.
+> DSC isn't implemented to process the `parameters` and `variables` properties yet. They can be
+> defined in a document, but not referenced.
+
+## Defining a configuration document
+
+Minimally, a configuration document defines the `$schema` and `resources` properties. The
+`$schema` property must be a valid URI for DSC's configuration document schema. The `resources`
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:49]:
```suggestion
`$schema` property must be a valid URI for the DSC configuration document schema. The `resources`
```

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md [hunk 3]
```diff
@@ -6,7 +6,201 @@ description: >-
...
+    type: Microsoft.Windows/Registry
+    properties:
+      keyPath:   HKCU\example\key
+      valueName: Example
+      valueData:
+        String: This is an example.
+```
+
+The example document declares a single resource instance called `example key value`. The instance
+uses the `Microsoft.Windows/Registry` resource to declare that:
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:68]:
```suggestion
The example document defines a single resource instance called `example key value`. The instance
uses the `Microsoft.Windows/Registry` resource to declare the following desired state:
```

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md [hunk 4]
```diff
@@ -6,7 +6,201 @@ description: >-
...
+
+The example document is _declarative_. It describes the desired state, not how to put the system
+into that state. It relies on the `Microsoft.Windows/Registry` resource to handle getting, testing,
+and setting the state of the registry key and value.
+
+### Defining resource instances
+
+As shown in the prior example, configuration documents include a collection of resource instances.
+Together, the instances describe a system's desired state. A configuration document can include any
+number of resource instances. The instances can be of any available DSC Resource.
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:82]:
```suggestion
Together, the instances describe the desired state of a system. A configuration document can include any
number of resource instances.
```

I am not sure what that last sentence (removed) was supposed to mean.

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md [hunk 5]
```diff
@@ -6,7 +6,201 @@ description: >-
...
+
+A resource instance declaration always includes:
+
+- `name` - A short, human-readable name for the instance that's unique in the document. This name
+  is used for logging and it helps to document an instance's purpose in the document.
+- `type` - The [fully qualified type name][02] for the resource to identify the resource DSC should
+  use to manage the instance.
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:89]:
```suggestion
  is used for logging and it helps describe the purpose of the instance.
- `type` - The [fully qualified type name][02] of the resource that DSC should use to manage the instance.
```

Are these 2 uses of "instance" talking about the same thing?

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md [hunk 6]
```diff
@@ -6,7 +6,201 @@ description: >-
...
+      $id: https://developer.microsoft.com/json-schemas/windows/registry/20230303/Microsoft.Windows.Registry.schema.json
+      keyPath: ''
+messages: []
+hadErrors: false
+```
+
+## Testing a configuration
+
+The `dsc config test` command compares the current state of the resource instances defined in a
+configuration document to their desired state. The result for each instance includes:
```
Comment (sdwheeler) [inline dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:122]:
```suggestion
The `dsc config test` command compares the current state of the resource instances to their desired state. The result for each instance includes:
```

**Review by sdwheeler — CHANGES_REQUESTED** [review]
I have questions

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md (hunk 4 above)
Comment (michaeltlombardi) [reply dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:82]: Trying to clarify that you're not limited to defining instances of a single resource type at a time - agree with cutting though.

File: dsc/docs-conceptual/dsc-3.0/concepts/configurations.md (hunk 5 above)
Comment (michaeltlombardi) [reply dsc/docs-conceptual/dsc-3.0/concepts/configurations.md:89]: Yes - the _instance_ is the arbitrary component being managed. It's defined in three parts:

- The `properties` keyword defines the desired state of the instance.
- The `type` keyword defines the resource that processes the instance. The processing includes both property validation and operation invocation.
- The `name` keyword defines an annotation for human readers and users of the configuration. The name is included in logging for the instance and serves as minimal documentation for the instance.

**Review by sdwheeler — APPROVED** [review]

**Pull request merged**